File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __help(self, root, ans):
        if root == None:
            return ans
        else:
            ans.append(root.val)
            self.__help(root.left, ans)
            self.__help(root.right, ans)

    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        self.__help(root, ans)

        return ans

File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int t = 0;

void __help(struct TreeNode* root, int *ans) {
    if (root == NULL) return;
    else {
        ans[t++] = root->val;
        __help(root->left, ans);
        __help(root->right, ans);
    }
}

int* preorderTraversal(struct TreeNode* root, int* returnSize) {
    int *ans;
    int k = 0;
    ans = (int*)malloc(101*sizeof(int));
    __help(root, ans);
    *returnSize = t;
    t = k;
    return ans;
}